proposals/1859_github-auth/README.md

```diff
@@ -0,0 +1,527 @@
+      this is assuming that the related bug in the Pinniped CLI is fixed as outlined in
+      [this PR](https://github.com/vmware-tanzu/pinniped/pull/1857).
+    - It should take care to correctly handle unicode characters that might exist in flag values when
+      invoking the subprocess, e.g. emojis in the upstream IDP name flag.
+      As long as the subprocess is successful, its stdout results can be ignored, because we only care about
+      what it cached in the session cache as a side effect.
+4. Now that the user is authenticated, the command will need to be able to construct a session cache key
+   exactly the same way that the "pinniped login oidc" subprocess did it, so it can be sure that it will be
+   looking up the same session in the cache.
+5. Next, it could open the session cache and read the access token from it. This access token should still be
+   valid/non-expired because of the "pinniped login oidc" subprocess that just succeeded.
+6. Finally, it could call the new Supervisor API to add consent for the PAT by sending the access token
+   for authentication and sending the SHA-256 hash of the PAT to add it to the allow list for the current user
+   identified by the access token. Revoking consent for a specific PAT would be a similar request,
+   and revoking consent for all PATs associated with the current user could work in a similar way
+   (but without needing to submit any hashed value of a specific PAT).
+
+#### Upgrades
+
+All changes will be backwards-compatible additions.
+
+#### Tests
+
+All aspects of the new features will be unit tested. Appropriate integration tests will also be added.
+
+#### New Dependencies
+
+We may like to use Google's golang client package for GitHub to help us call the GitHub API, although this would not
+be strictly necessary since these are simple REST API calls. It may help us implement things like pagination.
+
+#### Performance Considerations
+
+None.
+
+#### Observability Considerations
+
+Follow our pre-existing standards for error messages, log messages, custom resource status, etc.
+
+#### Security Considerations
+
+Some GitHub-specific considerations are already discussed above. Otherwise, this fits into the existing design
+of Pinniped without changing any of its existing security considerations.
+
+#### Usability Considerations
+
+No significant changes for end users in terms of how they authenticate or use kubectl.
+
+#### Documentation Considerations
+
+We will add API docs for GitHubIdentityProvider, and add docs on the website for authenticating with GitHub.
+
+### Other Approaches Considered
+
+None.
+
+## Open Questions
+
+- Needs investigation: How does the GitHub teams API deal with membership in nested teams? Does it flatten them for us?
+- Needs investigation: Does the GitHub API return org names with capital letters?
+  e.g. how does it return this org name? https://github.com/Broadcom (note the capital "B").
```

Comment:
This organization is uniquely identified with `login: Broadcom` with a capital B.